File: 프로그래머스/lv2/임강민/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    while storey != 0:
        current = storey % 10
        if current > 5:
            answer += 10 - current
            storey += 10
        elif current < 5:
            answer += current
        else:

        storey //= 10
    return answer
    # answer = 0
    # c = 8
    # while storey > 0:
    #     if storey // (10**c) >= 1:
    #         storey -= 10**c
    #         answer += 1
    #     else:
    #         c -= 1
    # return answer

File: 프로그래머스/lv2/김유현/마법의 엘리베이터.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int storey) {
    int answer = 0;
    
    while (storey) {
        // 현재 자릿값 
        int number = storey % 10;
    
        if (number <= 5) {
            answer += number; // 4 -> 3 -> 2 - > 1 -> 0
        } else if (number > 5) {
            answer += 10 - number; // 6 -> 7 -> 8 -> 9 -> 10 
            storey += 10;
        }
            
        storey /= 10;
    }
    

    return answer;
}